int32 mode
int32 reference

